File: sdkwork-backend-sdk-csharp/Api/CardTemplateApi.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Backend.Http;
using Backend.Models;

namespace Backend.Api
{
    public class CardTemplateApi
    {
        private readonly HttpClient _client;

        public CardTemplateApi(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// Update an existing card template
        /// </summary>
        public async Task<PlusApiResultPlusCardTemplateVO?> UpdateAsync(PlusCardTemplateForm body)
        {
            return await _client.PutAsync<PlusApiResultPlusCardTemplateVO>(ApiPaths.BackendPath("/card/template"), body);
        }

        /// <summary>
        /// Create a new card template
        /// </summary>
        public async Task<PlusApiResultPlusCardTemplateVO?> CreateAsync(PlusCardTemplateForm body)
        {
            return await _client.PostAsync<PlusApiResultPlusCardTemplateVO>(ApiPaths.BackendPath("/card/template"), body);
        }

        /// <summary>
        /// Get card templates by page
        /// </summary>
        public async Task<PlusApiResultPagePlusCardTemplateVO?> ListByPageAsync(QueryListForm? body = null, Dictionary<string, object>? query = null)
        {
            return await _client.PostAsync<PlusApiResultPagePlusCardTemplateVO>(ApiPaths.BackendPath("/card/template/list"), body, query);
        }

        /// <summary>
        /// Get all card templates
        /// </summary>
        public async Task<PlusApiResultListPlusCardTemplateVO?> ListAllEntitiesAsync(QueryListForm? body = null)
        {
            return await _client.PostAsync<PlusApiResultListPlusCardTemplateVO>(ApiPaths.BackendPath("/card/template/list/all"), body);
        }

        /// <summary>
        /// Get a card template by ID
        /// </summary>
        public async Task<PlusApiResultPlusCardTemplateVO?> GetByIdAsync(string id)
        {
            return await _client.GetAsync<PlusApiResultPlusCardTemplateVO>(ApiPaths.BackendPath($"/card/template/{id}"));
        }

        /// <summary>
        /// Delete a card template
        /// </summary>
        public async Task<PlusApiResultBoolean?> DeleteAsync(string id)
        {
            return await _client.DeleteAsync<PlusApiResultBoolean>(ApiPaths.BackendPath($"/card/template/{id}"));
        }
    }
}

File: sdkwork-backend-sdk-swift/Sources/API/CardTemplateApi.swift
import Foundation

public class CardTemplateApi {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Update an existing card template
    public func update(body: PlusCardTemplateForm) async throws -> PlusApiResultPlusCardTemplateVO? {
        let response = try await client.put(ApiPaths.backendPath("/card/template"), body: body)
        return response as? PlusApiResultPlusCardTemplateVO
    }

    /// Create a new card template
    public func create(body: PlusCardTemplateForm) async throws -> PlusApiResultPlusCardTemplateVO? {
        let response = try await client.post(ApiPaths.backendPath("/card/template"), body: body)
        return response as? PlusApiResultPlusCardTemplateVO
    }

    /// Get card templates by page
    public func listByPage(body: QueryListForm? = nil, params: [String: Any]? = nil) async throws -> PlusApiResultPagePlusCardTemplateVO? {
        let response = try await client.post(ApiPaths.backendPath("/card/template/list"), body: body, params: params)
        return response as? PlusApiResultPagePlusCardTemplateVO
    }

    /// Get all card templates
    public func listAllEntities(body: QueryListForm? = nil) async throws -> PlusApiResultListPlusCardTemplateVO? {
        let response = try await client.post(ApiPaths.backendPath("/card/template/list/all"), body: body)
        return response as? PlusApiResultListPlusCardTemplateVO
    }

    /// Get a card template by ID
    public func getById(id: String) async throws -> PlusApiResultPlusCardTemplateVO? {
        let response = try await client.get(ApiPaths.backendPath("/card/template/\(id)"))
        return response as? PlusApiResultPlusCardTemplateVO
    }

    /// Delete a card template
    public func delete(id: String) async throws -> PlusApiResultBoolean? {
        let response = try await client.delete(ApiPaths.backendPath("/card/template/\(id)"))
        return response as? PlusApiResultBoolean
    }
}
